code/modules/events/meaty_gore.dm

```diff
@@ -12,3 +14,66 @@
 
 /datum/event/meteor_wave/gore/end()
 	GLOB.minor_announcement.Announce("The station has cleared the debris.", "Debris Alert")
+
+	/***************************************\
+	|*********Stowaway Changelings***********|
+	\***************************************/
+
+/datum/event/meteor_wave/gore/ling
+	var/changelings_amount
+	var/list/candidates
+
+	var/spawncount
+
+/datum/event/meteor_wave/gore/ling/announce()
+	return //TSS SECRET
+
+/datum/event/meteor_wave/gore/ling/setup()
+	changelings_amount = round(TGS_CLIENT_COUNT / 40) //120 players for 3 changelings
+	if(changelings_amount < 1 && TGS_CLIENT_COUNT >= NEED_PLAYERS)
+		changelings_amount = 1
+	spawncount = changelings_amount
+	var/Need = NEED_PLAYERS
+
+	var/image/img = image('icons/mob/actions/actions.dmi', "chameleon_outfit")
+	candidates = SSghost_spawns.poll_candidates("Do you want to play as a Stowaway Changeling?", ROLE_CHANGELING, TRUE, source = img, poll_time = 40 SECONDS)
+
+	if(!length(candidates) || TGS_CLIENT_COUNT < NEED_PLAYERS || GAMEMODE_IS_CULT || GAMEMODE_IS_NUCLEAR || GAMEMODE_IS_REVOLUTION) //i just love idea of gamemodes combination, but boring contributors (and some balancers guys) says no...
+		var/the_message = "Stowaway Changelings had no volunteers ([length(candidates)]) or gamemode ([SSticker.mode.name]) is not valid or is too few clients (have [TGS_CLIENT_COUNT] Vs need [Need]) on server and was canceled."
+		message_admins("[the_message]")
+		log_admin("[the_message]")
+
+		for(var/mob/dead/observer/O in GLOB.player_list)
+			if(O.client)
+				to_chat(O, "<span class='ghostalert'>[the_message]</span>")
+
+		//just imagine... CULTIST CHANGELING... NUKIES AND CHANGELINGS... REVOLUTION CHANGELING... eh... SHADOWLING THRALL (yep they're deleted) CHANGELING MUHAHAHAHAHHAHA
+		var/datum/event_container/EC = SSevents.event_containers[EVENT_LEVEL_MODERATE] //no lings? okay, go and get compensation!
+		EC.next_event_time = world.time + (60 * 10)
+		return kill()
```

Comment:
```suggestion
	if(!length(candidates) || TGS_CLIENT_COUNT < NEED_PLAYERS || GAMEMODE_IS_CULT || GAMEMODE_IS_NUCLEAR || GAMEMODE_IS_REVOLUTION) //i just love idea of gamemodes combination, but boring contributors (and some balancers guys) says no...
		var/the_message = "Stowaway Changelings had no volunteers ([length(candidates)]) or gamemode ([SSticker.mode.name]) is not valid or is too few clients (have [TGS_CLIENT_COUNT] Vs need [Need]) on server and was canceled."
		message_admins("[the_message]")
		log_admin("[the_message]")

		for(var/mob/dead/observer/O in GLOB.player_list)
			if(O.client)
				to_chat(O, "<span class='ghostalert'>[the_message]</span>")

		//just imagine... CULTIST CHANGELING... NUKIES AND CHANGELINGS... REVOLUTION CHANGELING... eh... SHADOWLING THRALL (yep they're deleted) CHANGELING MUHAHAHAHAHHAHA
		var/datum/event_container/EC = SSevents.event_containers[EVENT_LEVEL_MODERATE] //no lings? okay, go and get compensation!
		EC.next_event_time = world.time + (60 * 10)
		return kill()
```
this should fail quietly if it's the wrong gamemode. use length(get_living_players(exclude_nonhuman = FALSE, exclude_offstation = TRUE) again and don't cast NEED_PLAYERS to a var